managed/models/database.go

```diff
@@ -1058,36 +1061,64 @@ func setupFixture1(q *reform.Querier, username, password string) error {
 	if _, err = CreateNodeExporter(q, PMMServerAgentID, nil, false, []string{}, nil, ""); err != nil {
 		return err
 	}
+	address, port, err := preparePGAddress(params.Address)
+	if err != nil {
+		return err
+	}
+	if params.Address != DefaultPostgreSQLAddr {
+		if node, err = CreateNode(q, RemoteNodeType, &CreateNodeParams{
+			NodeName: "pmm-server-db",
+			Address:  address,
+		}); err != nil {
+			return err
+		}
+	}
 
 	// create PostgreSQL Service and associated Agents
 	service, err := AddNewService(q, PostgreSQLServiceType, &AddDBMSServiceParams{
 		ServiceName: PMMServerPostgreSQLServiceName,
 		NodeID:      node.NodeID,
-		Address:     pointer.ToString("127.0.0.1"),
-		Port:        pointer.ToUint16(5432),
+		Database:    params.Name,
+		Address:     &node.Address,
+		Port:        pointer.ToUint16(port),
 	})
 	if err != nil {
 		return err
 	}
-	_, err = CreateAgent(q, PostgresExporterType, &CreateAgentParams{
-		PMMAgentID: PMMServerAgentID,
-		ServiceID:  service.ServiceID,
-		Username:   username,
-		Password:   password,
-	})
+
+	ap := &CreateAgentParams{
+		PMMAgentID:    PMMServerAgentID,
+		ServiceID:     service.ServiceID,
+		TLS:           params.SSLMode != "disable",
+		TLSSkipVerify: params.SSLMode == "disable" || params.SSLMode == "verify-ca",
+		Username:      params.Username,
+		Password:      params.Password,
+	}
+	if ap.TLS {
+		ap.PostgreSQLOptions = &PostgreSQLOptions{}
+		for path, field := range map[string]*string{
+			params.SSLCAPath:   &ap.PostgreSQLOptions.SSLCa,
+			params.SSLCertPath: &ap.PostgreSQLOptions.SSLCert,
+			params.SSLKeyPath:  &ap.PostgreSQLOptions.SSLKey,
+		} {
+			if len(path) == 0 {
+				continue
+			}
+			content, err := os.ReadFile(path) //nolint:gosec
+			if err != nil {
+				return err
+			}
+			*field = string(content)
+		}
+	}
```

Comment:
Shouldn't we give user the option to not cache certs?

Reply:
This case is not described in the task. I wonder what @atymchuk and @BupycHuk think about this.

Reply:
what do you mean by not cache certs? we store custom certs to be able to connect to PostgreSQL with custom certs.
BTW @rishat-ishbulatov it's better to use `path == ""` instead of `len(path) == 0`